### PR TITLE
Develop pre check

### LINKED ISSUE #37957: STYLE: fail on pd.testing direct usage
`pd.testing.assert_series_equal(result, expected)`

should fail style checks, rather we want `tm.assert_series_equal`

we need to update any usages in the codebase as well as add this to the pre-commit checks.

### PR DESCRIPTION
**Pull request by cjlynch278**
- [ x] closes #37957
- [ ] tests added / passed
- [ x passes `black pandas`
- [ x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -154,6 +154,11 @@ repos:
             pandas/tests/io/excel/test_writers\.py|
             pandas/tests/io/pytables/common\.py|
             pandas/tests/io/pytables/test_store\.py$
+    -   id: check-no-pd-class
+        name: check for pd class.
+        types: [python]
+        entry: pd.testing.assert_series_equal
```
Comment (jreback) [inline .pre-commit-config.yaml:160]: this needs to be pd.testing.* (not exatcly sure how to do the wildcard here). cc @MarcoGorelli

Comment (cjlynch278) [reply .pre-commit-config.yaml:160]: ah I see, thanks for the feed back. If my knowledge is correct, a wild card is unnecessary. I'll update this.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -154,6 +154,11 @@ repos:
             pandas/tests/io/excel/test_writers\.py|
             pandas/tests/io/pytables/common\.py|
             pandas/tests/io/pytables/test_store\.py$
+    -   id: check-no-pd-class
+        name: check for pd class.
+        types: [python]
+        entry: pd.testing.
```
Comment (jreback) [inline .pre-commit-config.yaml:160]: suspect need to escape `\` the dots

**Review by jreback — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jreback) [reply .pre-commit-config.yaml:160]: cc @MarcoGorelli

**Comment by MarcoGorelli** [comment]
Nice, thanks @cjlynch278 !

Could this fit into the existing `non-standard-imports-in-tests` check?

And yes, you'll need to escape dots (`pygrep` uses `re.search`)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Nice, thanks @cjlynch278 !

Could this fit into the existing `non-standard-imports-in-tests` check? Perhaps it could be renamed to `check-unwanted-patterns-in-tests`

And yes, you'll need to escape dots (`pygrep` uses `re.search`)

**Comment by cjlynch278** [comment]
Hey @MarcoGorell and @jreback I added my changes to the existing check, implementing a wildcard. This PR should be ready for final review.

In addition, I think a great approach to implementing the pre-commit library would be to not only identify problems like we do now, but to fix them automatically. We can do this using the search and replace plugin - https://github.com/mattlqx/pre-commit-search-and-replace.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -89,6 +89,9 @@ repos:
             # No direct imports from conftest
             conftest\ import|
             import\ conftest
+
+            # Check for use of pandas.testing instead of tm
+            pd.testing.\
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:94]: You need to escape the dots
```suggestion
            pd\.testing\.
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @cjlynch278 !

**Pull request merged**